tests/console/yast2_ftp.pm

```diff
@@ -26,108 +27,95 @@ sub run() {
-    send_key 'down';
+    wait_screen_change { send_key 'down' };
     wait_still_screen 1;
-    send_key 'ret';
+    wait_screen_change { send_key 'ret' };
     wait_still_screen 1;
-    send_key 'alt-e';
+    wait_screen_change { send_key 'alt-e' };
     assert_screen 'yast2_ftp_authentication_enabled';
-    send_key 'alt-y';
+    wait_screen_change { send_key 'alt-y' };
     wait_still_screen 1;
-    send_key 'alt-s';
-    send_key 'alt-s';                              # disable creating directories
-    assert_screen 'yast2_ftp_anonymous_upload';    # check upload settings
+    send_key_until_needlematch 'yast2_tftp_anon_create_dir_disabled', 'alt-s';     # disable creating directories
+    assert_screen 'yast2_ftp_anonymous_upload';                                    # check upload settings
 
     # Expert Settings
     send_key_until_needlematch 'yast2_tftp_authentication_selected', 'shift-tab';
-    send_key 'down';
-    wait_still_screen 1;
-    send_key 'ret';
-    wait_still_screen 1;
-    assert_screen 'yast2_ftp_create_upload_dir_confirm';    # confirm to create upload directory
-    send_key 'alt-y';
-    wait_still_screen 1;
-    send_key 'alt-m';
-    for (1 .. 4) { send_key 'down'; }
-    send_key 'alt-a';
-    for (1 .. 4) { send_key 'up'; }
-    send_key 'alt-l';                                       # enable SSL
-    assert_screen 'yast2_ftp_expert_settings';              # check passive mode value and enable SSL
-    send_key 'alt-s';                                       # give path for DSA certificate
+    wait_screen_change { send_key 'down' };
+    wait_screen_change { send_key 'ret' };
+    assert_screen 'yast2_ftp_create_upload_dir_confirm';                           # confirm to create upload directory
+    wait_screen_change { send_key 'alt-y' };
+    wait_screen_change { send_key 'alt-m' };
+    type_string_slow "30000\n";
+    wait_screen_change { send_key 'alt-a' };
+    type_string_slow "30100\n";
+    wait_screen_change { send_key 'alt-l' };                                       # enable SSL
+    assert_screen 'yast2_ftp_expert_settings';                                     # check passive mode value and enable SSL
+    wait_screen_change { send_key 'alt-s' };                                       # give path for DSA certificate
     type_string '/etc/vsftpd.pem';
-    send_key 'alt-p';                                       # open port in firewall
-    wait_still_screen 1;
-    send_key 'alt-f';                                       # done and close the configuration page now
-
+    wait_screen_change { send_key 'alt-p' };                                       # open port in firewall
+    wait_screen_change { send_key 'alt-f' };                                       # done and close the configuration page now
 
     # yast might take a while on sle12 due to suseconfig
-    wait_serial("yast2-ftp-server-status-0", 60) || die "'yast2 ftp-server' didn't finish";
+    assert_screen 'root-console', timeout => 180;
```

Comment:
One of the issues here is that script_run shows red step in execution if there was no output on serial, so I would still stick to type_string here. In general test fails on leap mainly because some steps break timeouts...